lib/fluent/root_agent.rb

```diff
@@ -54,17 +55,30 @@ def initialize(log:, system_config: SystemConfig.new)
       @inputs = []
       @suppress_emit_error_log_interval = 0
       @next_emit_error_log_time = nil
-      @without_source = false
-      @enable_input_metrics = false
+      @without_source = system_config.without_source || false
+      @source_only_buffer_agent = nil
+      @enable_input_metrics = system_config.enable_input_metrics || false
+
+      @with_source_only = false
+      if system_config.with_source_only
+        if Fluent.windows?
+          log.warn "with-source-only is disabled because it is not supported on Windows"
```

Comment:
Thanks!
It may be better to raise an Exception here.

If you are talking about not running the `config_param` in SystemConfig on Windows, that control would not be necessary. 

Because...

* About command line options, Fluentd doesn't start with nonexistent options.
* On the other hand, about configs, Fluentd ignores nonexistent settings.

So, there is no actual example of such a thing, at least not in this repository.

Example:

https://github.com/fluent/fluentd/blob/eebc7e1da6e0916be2ccdfab79fe0cffef296d21/lib/fluent/plugin/out_forward.rb#L213-L220

https://github.com/fluent/fluentd/blob/eebc7e1da6e0916be2ccdfab79fe0cffef296d21/lib/fluent/plugin/out_file.rb#L171-L174

So, we should consider whether to disable it or stop starting Fluentd when it is set on Windows.

Reply:
Fixed to raise ConfigError on Windows.